src/amo/components/SearchForm.js

```diff
@@ -25,10 +31,10 @@ export class SearchFormBase extends React.Component {
   }
 
   goToSearch(query) {
-    const { api, pathname } = this.props;
+    const { addonType, api, pathname } = this.props;
     this.context.router.push({
       pathname: `/${api.lang}/${api.clientApp}${pathname}`,
-      query: { q: query },
+      query: { q: query, type: addonType },
```

Comment:
`addonType` is already used by the search code to limit the scope of add-on type returned by the API, so yes it's already used and there is already code and tests written for it. That's why this patch adds so little; we're just using that already supported feature for filtering by `addonType` to also inform the current view context.

We don't have values for `viewContext` that *arent'* addonType, and we would need to go back and change lots of search code to change this. So I don't think this should change to `viewContext`